packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -297,31 +298,38 @@ export async function openPS(
 
 export function getDataSetTypeAndOptions(type: string) {
     let typeEnum;
+    let cliDefaultsKey;
     let createOptions;
     switch (type) {
         case localize("createFile.dataSetBinary", "Data Set Binary"):
             typeEnum = zowe.CreateDataSetTypeEnum.DATA_SET_BINARY;
+            cliDefaultsKey = "BINARY";
             createOptions = vscode.workspace.getConfiguration("Zowe-Default-Datasets-Binary");
             break;
         case localize("createFile.dataSetC", "Data Set C"):
             typeEnum = zowe.CreateDataSetTypeEnum.DATA_SET_C;
+            cliDefaultsKey = "C";
             createOptions = vscode.workspace.getConfiguration("Zowe-Default-Datasets-C");
             break;
         case localize("createFile.dataSetClassic", "Data Set Classic"):
             typeEnum = zowe.CreateDataSetTypeEnum.DATA_SET_CLASSIC;
+            cliDefaultsKey = "CLASSIC";
             createOptions = vscode.workspace.getConfiguration("Zowe-Default-Datasets-Classic");
             break;
         case localize("createFile.dataSetPartitioned", "Data Set Partitioned"):
             typeEnum = zowe.CreateDataSetTypeEnum.DATA_SET_PARTITIONED;
+            cliDefaultsKey = "PARTITIONED";
             createOptions = vscode.workspace.getConfiguration("Zowe-Default-Datasets-PDS");
             break;
         case localize("createFile.dataSetSequential", "Data Set Sequential"):
             typeEnum = zowe.CreateDataSetTypeEnum.DATA_SET_SEQUENTIAL;
+            cliDefaultsKey = "SEQUENTIAL";
```

Comment:
Do you think it is good to put this in globals? Maybe we can re-use it in future enhancements?

Reply:
I am not sure if this is useful...the `cliDefaultsKey` is just the `CreateDataSetTypeEnum`, but without `DATA_SET_`

ie. if the `CreateDataSetTypeEnum` is `DATA_SET_BINARY`, then the `cliDefaultsKey` is `BINARY`...

So it feels a little pointless to me to make a whole new enum just to remove the `DATA_SET_` part of `CreateDataSetTypeEnum`

The CLI is irritating me a bit with all their different names for the same things. `CreateDataSetTypeEnum` should just match the keys used in the CLI to identify the default settings (keys used in `zowe.CreateDefaults` file). We are having the same issue with sending profile values to Imperative as CLI commands. I think they need to standardize their naming procedures, and we shouldn't be making a bunch of different data structures that only have slight differences in their values to accommodate all these slightly different names across CLI and Imperative.

Rant over. Maybe I will bring this up in standup.

Reply:
ok. I understand. Do you think trim would be useful since you know that you want to remove DATA_SET_? like this:

const session = node.label.trim().substring(node.label.trim().indexOf("[") + 1, node.label.trim().indexOf("]"));

Reply:
There's an example in C:\Users\jp669971\Documents\GitHub\Community\vscode-extension-for-zowe\packages\zowe-explorer\src\dataset\actions.ts. Line 989

Reply:
The problem is that I can't get the key from the enum. Ex. I don't have DATA_SET_SEQUENTIAL as a string, I have just the number: 4.

If you look here, since it is a `const enum`, the key is removed completely on compilation...so it doesn't even exist there for me to get ahold of: [link to docs](https://www.typescriptlang.org/docs/handbook/enums.html#const-enums)

Maybe it makes sense to add it as an enum for this reason, but I still think it's a bit overkill P:

Reply:
I could add an enum like this:
```
CreateDataSetTypeKeysEnum {
    0 = DATA_SET_BINARY,
    1 = DATA_SET_C,
    2 = DATA_SET_CLASSIC,
    3 = DATA_SET_PARTITIONED,
    4 = DATA_SET_SEQUENTIAL
}

```
And that would at least be more useful, in that we can take the `CreateDataSetTypeEnum` and reverse it to get the keys when we need them...but it is sort of a hack to get around their `const enum` definition, and it makes me wonder: why did they define it as `const enum` in the first place? Maybe they can remove it? Maybe we shouldn't go around it?

Reply:
I think that "hack" is the approach that I'm gonna go for. But only if you find it very useful. 

> why did they define it as const enum in the first place? 
That's a good question. We can open an enhancement issue for this. I pretty sure they didn't expect anybody to get the description of the dataset. So this might be the first time.